### PR TITLE
Addition of multiplierFESpace

### PR DESCRIPTION
**Pull request by lokmanb**
This PR introduces a first development of MultiplierFESpace for the solution of constrained minimization problems. The idea is to be able to construct a MultiFESpace combining the usual finite element discretization on the one hand and Lagrange multipliers on the other. Here the Lagrange multipliers are additional unknowns and the MultiFESpace combined with MultiplierFESpace enables to assemble a global system which yields the solution of the constrained minimization problem as well as the associated Lagrange multipliers. The development is built on the constrained Poisson example.

### CODE REVIEW

File: src/fespace/fespace.jl [hunk 1]
```diff
@@ -296,6 +296,26 @@ function TrialFESpace(
...
+"""
+function MultiplierFESpace(mesh::AbstractMesh, size::Int = 1, kwargs...)
+    fSpace = FunctionSpace(:Lagrange, 0)
+
+    iglob = collect(1:size)
+    offset = zeros(ncells(mesh), size)
+    for i in 1:size
+        offset[:, i] .= i - 1
+    end
+    ndofs = ones(ncells(mesh), size) # Temporary allocation for development purposes
```
Comment (bmxam) [inline src/fespace/fespace.jl:311]: can you remove the comment since it seems that we finally kept this line

File: src/assembler/assembler.jl [hunk 2]
```diff
@@ -456,6 +456,20 @@ function _count_n_elts(
     return n
 end
 
+"""
```
Comment (bmxam) [inline src/assembler/assembler.jl:459]: Can you please merge the three `_count_n_elts` doc strings into a unique one, located at the top of the first method declaration? Take a look at `src/fespace/fespace.jl` -> `TrialFESpace` constructor for inspiration.

**Review by bmxam — CHANGES_REQUESTED** [review]
We need some unit tests, especially on the dofhandler for the MultiplierFESpace. Maybe some tests taken from the tutorial configuration?

Moreover, a "global" test like Ghislain did for the DG Poisson would be nice. We can do it together during a live session.

**Comment by lokmanb** [comment]
Thanks for the review. I added a few unit tests for MultiplierFESpace. Concerning the "global" test, did you mean a test at the end of the corresponding example (with the error computation)?

**Comment by bmxam** [comment]
> Thanks for the review. I added a few unit tests for MultiplierFESpace. Concerning the "global" test, did you mean a test at the end of the corresponding example (with the error computation)?

I meant encapsulating your whole example (or a simplified version) into a `@testset` with some kind of test at the end (on the error for instance). Like it is done here for the Poisson DG : https://github.com/bcube-project/Bcube.jl/blob/d81437d097f86e5baa993c4eec45afca880d04fb/test/dof/test_assembler.jl#L213

**Review by bmxam — APPROVED** [review]

**Review by ghislainb — APPROVED** [review]

**Pull request merged**